[feature/client-pet-photo] Client 마이펫 탭 프로필 사진 API 구현

HanJuK opened the pull request:
## 개요
### PR 종류
**해당되는 한 가지만 선택하세요.**
 - [X] 기능 구현을 위한 소스코드 수정 (PR제목 양식: "[feature/브랜치명] 제목", 브랜치작명법: "feature/명칭" 사용)
 - [ ] 버그 수정을 위한 소스코드 수정 (PR제목 양식: "[hotfix/브랜치명] 제목", 브랜치작명법: "hotfix/명칭" 사용)
 - [ ] 소스코드 수정을 동반하지 않는 단순 문서 편집 (PR제목 양식: "[docs/브랜치명] 제목", 브랜치작명법: "docs/명칭" 사용)

### 작업내용
안드로이드 Client 마이펫 탭의 프로필 사진 API를 구현하였습니다.

## 변경사항
- 펫 프로필 사진 불러오기(fetch)와 수정하기(update) 구현
    - UpdatePetPhotoReqDto는 서버 측의 API에서 ModelAttribute과 MultipartFile을 사용하여 클라이언트 측에서 단순히 DTO 생성 후 보내는 데 어려움이 있어 request DTO 대신 Part를 이용하여 필요한 요소를 각각 독립적으로 보내는 방식으로 구현(ServerApi.kt의 updatePetPhotoReq() 함수 참고)
    - FetchPetPhotoResDto는 서버 측에서 API 호출 이후 리턴하는 값이 예외일 경우에는 DTO, 일반적인 경우에는 RAW 데이터(ByteArray)이기 때문에 클라이언트 측에서는 response DTO 대신 ResponseBody를 이용하는 방식으로 구현(fetchPetPhoto() 함수 참고)
    - 사용자로부터 입력받은 사진을 그대로 서버에 보낼 수 없기 때문에 inputStream과 outputStream을 이용하여 해당 파일을 복사한 후 보내는 방식으로 구현(이후 해당 파일은 삭제됨)
    -  서버 관련 함수를 저장하기 위해 ServerUtil.kt 이라는 파일을 생성(현재는 앞서 말한 파일 복사하는 함수만 존재)
- 펫 프로필 사진이 null 인 경우 보여줄 기본 프로필 사진 생성 및 sample 사진들 삭제
- 각종 오타 수정 및 리팩토링
    - updatePetResDto -> updatePetApiCall
    - image -> photo
    - ..등

## 비고
**펫 프로필 사진 불러오기(fetch)**
<img src="https://user-images.githubusercontent.com/48195650/128449977-d135ee58-b889-4be7-a35f-f0491d3fd6b3.png" width="40%"> <img src="https://user-images.githubusercontent.com/48195650/128449987-8a10778c-834a-4ee5-97d1-1cc519cb7029.png" width="40%">

**펫 프로필 사진 수정하기(update)**
<img src="https://user-images.githubusercontent.com/48195650/128450085-2e5b45ce-959e-43f2-8c8e-53dceca7ad39.png" width="40%"> <img src="https://user-images.githubusercontent.com/48195650/128450106-bc145b84-8ec8-4bd7-8f8a-805b30cda245.png" width="40%">

**저장된 펫 프로필 사진**
![s5](https://user-images.githubusercontent.com/48195650/128450278-73c83998-edc1-4c25-b15e-f13fea418ea0.PNG)


## 품질 관리를 위한 체크리스트
 - [ ] 해당사항 없음 (**소스코드 수정을 하지 않았습니다.**)
### 소스코드 테스트
**세 가지 테스트 모두를 실시하는 것을 권장하며, 최소한 작동확인(수동 테스트)은 해보셔야 합니다.**
 - [X] 수동 테스트 / Manual Test (앱 구동후 사용확인을 통한 직접 테스트 또는 서버 구동후 Postman을 이용한 테스트)
 - [ ] 자동화 단위 테스트 / Unit Test (JUnit 등을 통해 개별 단위로직에 대한 테스트 스크립트 작성)
 - [ ] 자동화 통합 테스트 / Integration Test (Spring MVC Testing 등을 통해 프로그램 전체에 대한 테스트 스크립트 작성)
### 기본 확인사항
 - [X] IDE에서 검출되는 모든 경고와 오류를 해결하셨습니까?
 - [X] 코딩 컨벤션(-추후 문서화 예정- 작명법, 금기사항, 구현방식 등)을 준수하셨습니까?
 - [X] 버그 또는 구현상의 실수, 누락 등 하자사항이 없는지 테스트하여 확인해보셨습니까?
 - [X] 명칭(변수명, 함수명, 주석 등)에 오타가 있는지 테스트하여 확인해보셨습니까?
 - [X] 이번 변경사항에 포함된 주석 및 기능 명세, 관련 문서를 최신화하셨습니까?

HanJuK committed:
logic for pet photo update API + typo fix(updatePetResDto -> updatePetApiCall)

HanJuK committed:
logic for fetching pet photos + lifecycle related bug fixes + image -> photo refactoring

HanJuK committed:
created default image + bug fixes for when photo selection is null + deleted sample images

HanJuK committed:
revert FileService directory(server)

krapie reviewed (commented):
fetchPetPhoto() 가 PetListAdapter와 PetProfileFragment 두 곳에 존재하고 있네요. 리스트에서 따로 사진 정보를 받아와야하고, 또 프로필 상세로 들어가면 또 사진 정보를 받아와야 하는 점 때문인 것으로 알고 있는데, 이 점을 개선할 수 있을 것 같습니다.

fetchPetPhoto()를 PetListAdapter에서만 실행하고 받아오는 Bitmap을 마이펫 탭에서만 임시로 저장해서 프로필 상세로 들어갈 때 임시저장한 Bitmap URL을 다른 데이터처럼 Intent에 담아 프로필 상세에서 다른 데이터들처럼 보여줄 수 있지 않을까요?

이렇게 하면 fetchPetPhoto()와 같은 API를 다시 호출하지 않을 수 있어서 서버에도 부담이 덜어지고, 클라이언트도 빠르게 넘어갈 수 있을 것 같습니다. (특히 사진 파일이기 때문에, 데이터가 커서 최대한 호출 횟수를 줄이는 것이 추후에도 좋아보입니다.)

HanJuK commented:
> fetchPetPhoto() 가 PetListAdapter와 PetProfileFragment 두 곳에 존재하고 있네요. 리스트에서 따로 사진 정보를 받아와야하고, 또 프로필 상세로 들어가면 또 사진 정보를 받아와야 하는 점 때문인 것으로 알고 있는데, 이 점을 개선할 수 있을 것 같습니다.
> 
> fetchPetPhoto()를 PetListAdapter에서만 실행하고 받아오는 Bitmap을 마이펫 탭에서만 임시로 저장해서 프로필 상세로 들어갈 때 임시저장한 Bitmap URL을 다른 데이터처럼 Intent에 담아 프로필 상세에서 다른 데이터들처럼 보여줄 수 있지 않을까요?
> 
> 이렇게 하면 fetchPetPhoto()와 같은 API를 다시 호출하지 않을 수 있어서 서버에도 부담이 덜어지고, 클라이언트도 빠르게 넘어갈 수 있을 것 같습니다. (특히 사진 파일이기 때문에, 데이터가 커서 최대한 호출 횟수를 줄이는 것이 추후에도 좋아보입니다.)

동의합니다. 말씀하신 방향으로 수정하겠습니다.

HanJuK committed:
optimize pet photo fetch(read description)

- fetchPetPhoto() function now only present inside PetListAdapter.kt
- used bitmaps and bytearrays for storing image files

HanJuK committed:
revert FileService directory(server)
